cmd/go-yaml/node.go

```diff
@@ -53,9 +64,50 @@ func FormatNode(n yaml.Node) *NodeInfo {
 		}
 	}
 
+	// Handle StreamNode-specific fields
+	if info.Kind == "Stream" {
+		if n.Encoding != 0 {
+			info.Encoding = formatEncoding(n.Encoding)
+		}
+		if n.Version != nil {
+			info.Version = formatVersion(n.Version)
+		}
+		if len(n.TagDirectives) > 0 {
+			info.TagDirectives = make([]TagDirectiveInfo, len(n.TagDirectives))
+			for i, td := range n.TagDirectives {
+				info.TagDirectives[i] = TagDirectiveInfo{
+					Handle: td.Handle,
+					Prefix: td.Prefix,
+				}
+			}
+		}
+	}
+
 	return info
 }
 
+// formatEncoding converts an encoding constant to its string representation.
+func formatEncoding(e yaml.Encoding) string {
+	switch e {
+	case yaml.EncodingUTF8:
+		return "UTF-8"
+	case yaml.EncodingUTF16LE:
+		return "UTF-16LE"
+	case yaml.EncodingUTF16BE:
+		return "UTF-16BE"
+	default:
+		return "Any"
+	}
```

Comment:
I would use


```suggestion
func formatEncoding(e yaml.Encoding) string {
	switch e {
	case yaml.EncodingUTF8:
		return "UTF-8"
	case yaml.EncodingUTF16LE:
		return "UTF-16LE"
	case yaml.EncodingUTF16BE:
		return "UTF-16BE"
	case yaml.ANY_ENCODING:
		return "Any"
	default:
		return "Unknown"
	}
```

Also this could be moved as a Stringer of yaml.Encoding type